src/boot/paging_32.zig

```diff
@@ -61,7 +61,10 @@ export fn map_kernel_code_segment() void {
         page_table_entry.setAddr(addr);
         // Set page table entry.
         const p2_index = page_num + p2_index_offset;
-        p2_table[p2_index] = page_table_entry.entry;
+        // TODO: uncomment when Zig 32-bit codegen bug is fixed; use work-around for now.
```

Comment:
Looks like this is a bug in `ld`, not Zig. The object file produced by Zig contains the correct assembly. And the object file produced by `objcopy` also contains the correct assembly. But the assembly code produced by `ld` is trashed, such that the *high* 32-bits of `page_table_entry` when stored in `p2_table` are trashed (overwritten by the *low* 32-bits).